File: Days/Day07/01/01.py
#===== Declare Imports here if required =====


#===== Declare Global Variables / Functions here if required =====


class Node:
    def __init__(self, val):
        self.data = val
        self.next = None

def solve():
    N = int(input())
    arr = list(map(int, input().split()))

    #===== Declare Local Variables / Functions here if required =====


    head = None
    tail = None

    for x in arr:

        #===== Write Your Logic Here =====
        node = Node(x)
        if head == None:
            head = node
            tail = node
        else:
            tail.next = node
            tail = node

    e = head 
    while e != None:
        if e == head:
            print(f'{e.data}', end='')
        else:
            print(f' {e.data}', end='')
        e = e.next
        pass
if __name__ == "__main__":
    solve()

File: Days/Day07/01/01.cpp
#include<iostream>
using namespace std;

//===== Declare Imports here if required =====


//===== Declare Global Variables / Functions here if required =====


class Node{
public:
    int data;
    Node* next;
    Node(int val){
        data = val;
        next = NULL;
    }
};

//===== Declare Local Variables / Functions here if required =====


void solve(int N){

    Node* head = NULL;
    Node* tail = NULL;

    for(int i=0;i<N;i++){
        int x;
        cin >> x;

        //===== Write Your Logic Here =====
        Node* node = new Node(x);
        if(head == NULL) {
            head = node;
            tail = node;
        } else {
            tail->next = node;
            tail = node;
        }
    }

    //===== Write Your Logic Here =====
    Node* e = head;
    while(e != NULL) {
        if(e != head) {
            cout << " ";
        }
        cout << e->data;
        e = e->next;
    }

}

int main(){

    int N;
    cin >> N;

    solve(N);

    return 0;
}